tools/torch/data.lua

```diff
@@ -41,33 +41,205 @@ local function all_keys(cursor_,key_,op_)
         end)
 end
 
-local PreProcess = function(y, meanTensor, mirror, crop, train, cropY, cropX, croplen)
+-- HSV Augmentation
+-- Parameters:
+-- @param im (tensor): input image
+-- @param augHSV (table): standard deviations under {H,S,V} keys with float values.
+local augmentHSV = function(im_rgb, augHSV)
+    -- Fair augHSV standard deviation values are {H=0.02,S=0.04,V=0.08}
+    local im_hsv = image.rgb2hsv(im_rgb)
+    if augHSV.H >0.001 then
```

Comment:
Ah OK. Isn't it the responsibility of the user to choose a value that makes sense? I'm worried the magic value here could be confusing.